packages/squiggle-lang/src/library/registry/fnDefinition.ts

```diff
@@ -68,11 +94,20 @@ export function tryCallFnDefinition(
     }
     unpackedArgs.push(unpackedArg);
   }
+
+  // Fill in missing optional arguments with nulls.
+  // This is important, because empty optionals should be nulls, but without this they would be undefined.
+  if (unpackedArgs.length < fn.maxInputs) {
+    unpackedArgs.push(...Array(fn.maxInputs - unpackedArgs.length).fill(null));
+  }
+
   return fn.output.pack(fn.run(unpackedArgs, context));
 }
 
 export function fnDefinitionToString(fn: FnDefinition): string {
-  const inputs = fn.inputs.map((t) => t.getName()).join(", ");
+  const inputs = fn.inputs
+    .map((t) => t.getName() + (isOptional(t) && t.tag !== "named" ? "?" : ""))
```

Comment:
Ahh, I see. Yep, I don't see more elegant way to do this either.
There probably will be a better solution when we migrate `frTypes` to classes (then transparent or optional types would expose their underlying type in a public field and we could go around the type's `getName` to get it, or something).